Support mtl 2.3 and transformers 0.6

`microlens-mtl-0.2.0.1` shows a lot of errors when it's built with `mtl-2.3` and `--allow-newer`. For example:

```
src/Lens/Micro/Mtl/Internal.hs:402:94: error:
    • Variable not in scope:
        liftM
          :: (Either e0 a2 -> Err e0 a2) -> m (Either e c) -> m0 (Err e c)
    • Perhaps you meant one of these:
        ‘liftA’ (imported from Control.Applicative),
        ‘lift’ (imported from Control.Monad.State)
    |
402 |   zoom l = ExceptT . liftM getErr . zoom (\afb -> unfocusingErr #. l (FocusingErr #. afb)) . liftM Err . runExceptT
    |                                                                                              ^^^^^

src/Lens/Micro/Mtl/Internal.hs:525:36: error:
    • Variable not in scope:
        liftM2 :: (a1 -> a1 -> a1) -> m r -> m r -> m r
    • Perhaps you meant one of these:
        ‘liftA2’ (imported from Control.Applicative),
        ‘liftA’ (imported from Control.Applicative),
        ‘liftA3’ (imported from Control.Applicative)
    |
525 |   Effect ma <> Effect mb = Effect (liftM2 (<>) ma mb)
    |                                    ^^^^^^
```

It would be great to get a compatible release.

--- a/microlens-ghc/microlens-ghc.cabal
+++ b/microlens-ghc/microlens-ghc.cabal
@@ -1,5 +1,5 @@
 name:                microlens-ghc
-version:             0.4.13.1
+version:             0.4.13.2
 synopsis:            microlens + array, bytestring, containers, transformers
 description:
   Use this package instead of <http://hackage.haskell.org/package/microlens microlens> if you don't mind depending on all dependencies here – @Lens.Micro.GHC@ reexports everything from @Lens.Micro@ and additionally provides orphan instances of microlens classes for packages coming with GHC (<http://hackage.haskell.org/package/array array>, <http://hackage.haskell.org/package/bytestring bytestring>, <http://hackage.haskell.org/package/containers containers>, <http://hackage.haskell.org/package/transfromers transformers>).
@@ -44,7 +44,7 @@ library
                      , bytestring >=0.9.2.1 && <0.12
                      , containers >=0.4.0 && <0.7
                      , microlens ==0.4.12.*
-                     , transformers >=0.2 && <0.6
+                     , transformers >=0.2 && <0.7
 
   ghc-options:
     -Wall -fwarn-tabs

--- a/microlens-mtl/microlens-mtl.cabal
+++ b/microlens-mtl/microlens-mtl.cabal
@@ -1,5 +1,5 @@
 name:                microlens-mtl
-version:             0.2.0.1
+version:             0.2.0.2
 synopsis:            microlens support for Reader/Writer/State from mtl
 description:
   This package contains functions (like 'view' or '+=') which work on 'MonadReader', 'MonadWriter', and 'MonadState' from the mtl package.
@@ -39,8 +39,8 @@ library
   -- other-extensions:
   build-depends:       base >=4.5 && <5
                      , microlens >=0.4 && <0.5
-                     , mtl >=2.0.1 && <2.3
-                     , transformers >=0.2 && <0.6
+                     , mtl >=2.0.1 && <2.4
+                     , transformers >=0.2 && <0.7
                      , transformers-compat >=0.4 && <1
 
   ghc-options:

--- a/microlens-mtl/src/Lens/Micro/Mtl/Internal.hs
+++ b/microlens-mtl/src/Lens/Micro/Mtl/Internal.hs
@@ -51,6 +51,10 @@ where
 
 
 import Control.Applicative
+#if MIN_VERSION_mtl(2, 3, 0)
+import Control.Monad (liftM, liftM2)
+#else
+#endif
 import Control.Monad.Reader as Reader
 import Control.Monad.State as State
 import Control.Monad.Trans.State.Lazy as Lazy
@@ -59,9 +63,11 @@ import Control.Monad.Trans.Writer.Lazy as Lazy
 import Control.Monad.Trans.Writer.Strict as Strict
 import Control.Monad.Trans.RWS.Lazy as Lazy
 import Control.Monad.Trans.RWS.Strict as Strict
+#if !MIN_VERSION_transformers(0, 6, 0)
 import Control.Monad.Trans.Error
-import Control.Monad.Trans.Except
 import Control.Monad.Trans.List
+#endif
+import Control.Monad.Trans.Except
 import Control.Monad.Trans.Identity
 import Control.Monad.Trans.Maybe
 -- microlens
@@ -87,9 +93,11 @@ type instance Zoomed (Strict.RWST r w s z) = FocusingWith w z
 type instance Zoomed (Lazy.RWST r w s z) = FocusingWith w z
 type instance Zoomed (Strict.WriterT w m) = FocusingPlus w (Zoomed m)
 type instance Zoomed (Lazy.WriterT w m) = FocusingPlus w (Zoomed m)
+#if !MIN_VERSION_transformers(0, 6, 0)
 type instance Zoomed (ListT m) = FocusingOn [] (Zoomed m)
-type instance Zoomed (MaybeT m) = FocusingMay (Zoomed m)
 type instance Zoomed (ErrorT e m) = FocusingErr e (Zoomed m)
+#endif
+type instance Zoomed (MaybeT m) = FocusingMay (Zoomed m)
 type instance Zoomed (ExceptT e m) = FocusingErr e (Zoomed m)
 
 ------------------------------------------------------------------------------
@@ -386,16 +394,18 @@ instance (Monoid w, Zoom m n s t) => Zoom (Lazy.WriterT w m) (Lazy.WriterT w n) 
   zoom l = Lazy.WriterT . zoom (\afb -> unfocusingPlus #. l (FocusingPlus #. afb)) . Lazy.runWriterT
   {-# INLINE zoom #-}
 
+#if !MIN_VERSION_mtl(2, 3, 0) && !MIN_VERSION_transformers(0, 6, 0)
 instance Zoom m n s t => Zoom (ListT m) (ListT n) s t where
   zoom l = ListT . zoom (\afb -> unfocusingOn . l (FocusingOn . afb)) . runListT
   {-# INLINE zoom #-}
 
-instance Zoom m n s t => Zoom (MaybeT m) (MaybeT n) s t where
-  zoom l = MaybeT . liftM getMay . zoom (\afb -> unfocusingMay #. l (FocusingMay #. afb)) . liftM May . runMaybeT
-  {-# INLINE zoom #-}
-
 instance (Error e, Zoom m n s t) => Zoom (ErrorT e m) (ErrorT e n) s t where
   zoom l = ErrorT . liftM getErr . zoom (\afb -> unfocusingErr #. l (FocusingErr #. afb)) . liftM Err . runErrorT
+  {-# INLINE zoom #-}
+#endif
+
+instance Zoom m n s t => Zoom (MaybeT m) (MaybeT n) s t where
+  zoom l = MaybeT . liftM getMay . zoom (\afb -> unfocusingMay #. l (FocusingMay #. afb)) . liftM May . runMaybeT
   {-# INLINE zoom #-}
 
 instance Zoom m n s t => Zoom (ExceptT e m) (ExceptT e n) s t where

